container-engine-lib/lib/backend_impls/docker/docker_network_allocator/docker_network_allocator.go

```diff
@@ -142,14 +146,16 @@ func (provider *DockerNetworkAllocator) CreateNewNetwork(
 	)
 }
 
-// This algorithm only picks a network in the 172.16.1.0/24 - 17.16.0.255.0/24 list
+// This algorithm picks a network in the 172.16.0.0/16 range
 // https://github.com/hashicorp/serf/issues/385#issuecomment-208755148 - we try to follow RFC 6890
 // https://www.rfc-editor.org/rfc/rfc6890.html calls 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16 Private-Use (docker usually picks from 172.16.0.0/12)
-// Just with this range we can get 256 APICs; if we limit services per APIC to 256(/24)
-// For simplicity we limit it to 256 APICs and allow networks 172.16.1.0/24 - 17.16.255.0/24
+// We take IPs from the range 170.16.0.0/16, split equally into 2^6 networks, so the mask of a given network is:
+// 170.16.nnnnnn00.0/22 where nnnnnn are 6 bits to address the network, and the other 8 bits address the services
```

Comment:
are there 8 or 10 bits for services?! just checking; above we write 2^10 ips per network and here we wright 8 bits